Panic when meta refresh URL is single-quoted

Here's a simple test HTML file:

### `test.html`

```html
<!doctype html>
<html lang="en">
    <head>
        <title>ox-hugo Test Site</title>
        <meta http-equiv="refresh" content="0; URL='https://github.com/kaushalmodi/ox-hugo/issues'" />
    </head>
</html>
```

### How to make `htmltest` panic

```
htmltest test.html
```

### Log

```
panic: parse 'https://github.com/kaushalmodi/ox-hugo/issues': first path segment in URL cannot contain colon

goroutine 1 [running]:
github.com/wjdp/htmltest/output.CheckErrorPanic(0x78d580, 0xc0000a9dd0)
        /home/kmodi/go.apps/src/github.com/wjdp/htmltest/output/error.go:22 +0x86
github.com/wjdp/htmltest/htmldoc.NewReference(0xc0001c04d0, 0xc0001c08c0, 0xc00001cb87, 0x2f, 0xc00000dd20)
        /home/kmodi/go.apps/src/github.com/wjdp/htmltest/htmldoc/reference.go:34 +0x109
github.com/wjdp/htmltest/htmltest.(*HTMLTest).checkMetaRefresh(0xc0000ecb40, 0xc0001c04d0, 0xc0001c08c0)
        /home/kmodi/go.apps/src/github.com/wjdp/htmltest/htmltest/check-meta.go:31 +0x200
github.com/wjdp/htmltest/htmltest.(*HTMLTest).checkMeta(0xc0000ecb40, 0xc0001c04d0, 0xc0001c08c0)
        /home/kmodi/go.apps/src/github.com/wjdp/htmltest/htmltest/check-meta.go:12 +0x54
github.com/wjdp/htmltest/htmltest.(*HTMLTest).testDocument(0xc0000ecb40, 0xc0001c04d0)
        /home/kmodi/go.apps/src/github.com/wjdp/htmltest/htmltest/htmltest.go:218 +0x479
github.com/wjdp/htmltest/htmltest.Test(0xc0000a99e0, 0x1, 0x1, 0x49)
        /home/kmodi/go.apps/src/github.com/wjdp/htmltest/htmltest/htmltest.go:138 +0x91f
main.run(0xc0000a99e0, 0xc0000a99e0)
        /home/kmodi/go.apps/src/github.com/wjdp/htmltest/main.go:159 +0x1e8
main.main()
        /home/kmodi/go.apps/src/github.com/wjdp/htmltest/main.go:66 +0x298
```

--- a/htmltest/check-meta.go
+++ b/htmltest/check-meta.go
@@ -28,6 +28,15 @@ func (hT *HTMLTest) checkMetaRefresh(document *htmldoc.Document, node *html.Node
 		// Define ref from this
 		var ref *htmldoc.Reference
 		if len(contentSplit) == 2 {
+			if contentSplit[1][0]==34 || contentSplit[1][0]==39 {
+				hT.issueStore.AddIssue(issues.Issue{
+					Level:     issues.LevelError,
+					Message:   "url in meta refresh must not start with single or double quote",
+					Reference: ref,
+				})
+				return
+			}
+
 			ref = htmldoc.NewReference(document, node, contentSplit[1])
 		} else {
 			ref = htmldoc.NewReference(document, node, "")

--- a/htmltest/check-meta_test.go
+++ b/htmltest/check-meta_test.go
@@ -83,3 +83,10 @@ func TestMetaRefreshContentInvalid(t *testing.T) {
 	tExpectIssueCount(t, hT3, 1)
 	tExpectIssue(t, hT3, "invalid content attribute in meta refresh", 1)
 }
+
+// Fails when meta refresh url starts with a single or double quote
+func TestIssue92(t *testing.T) {
+	hT := tTestFile("fixtures/meta/issues/92.html")
+	tExpectIssueCount(t, hT, 2)
+	tExpectIssue(t, hT, "url in meta refresh must not start with single or double quote", 2)
+}

--- a/htmltest/fixtures/meta/issues/92.html
+++ b/htmltest/fixtures/meta/issues/92.html
@@ -1,0 +1,8 @@
+<!doctype html>
+<html lang="en">
+<head>
+    <title>ox-hugo Test Site</title>
+    <meta http-equiv="refresh" content="0; URL='https://github.com/kaushalmodi/ox-hugo/issues'" />
+    <meta http-equiv="refresh" content='0; URL="https://github.com/kaushalmodi/ox-hugo/issues"' />
+</head>
+</html>

